internal/pkg/cli/env_init.go

```diff
@@ -257,10 +261,21 @@ func (o *initEnvOpts) Ask() error {
 
 // Execute deploys a new environment with CloudFormation and adds it to SSM.
 func (o *initEnvOpts) Execute() error {
-	o.initRuntimeClients()
+	if err := o.initRuntimeClients(); err != nil {
+		return err
+	}
+	appVersion, err := o.appVersionGetter.Version()
+	if err != nil {
+		return fmt.Errorf("get template version of application %s: %w", o.appName, err)
+	}
+	if diff := semver.Compare(appVersion, version.LatestTemplateVersion()); diff > 0 && !o.allowAppDowngrade {
+		return &errCannotDowngradeAppVersion{
+			appName:    o.name,
+			appVersion: appVersion,
+		}
+	}
 	app, err := o.store.GetApplication(o.appName)
 	if err != nil {
-		// Ensure the app actually exists before we do a deployment.
```

Comment:
nit: I think we can keep this comment but modify it to "...before we write the manifest". I feel like the value of this comment is to prevent people from moving this code block to just before L279.